ambry-router/src/test/java/com/github/ambry/router/RouterTestHelpers.java

```diff
@@ -198,13 +197,14 @@ static void setServerErrorCodes(ServerErrorCode[] serverErrorCodesInOrder, Parti
   /**
    * Asserts that expected threads are not running after the router is closed.
    */
-  static void assertCloseCleanup(NonBlockingRouter router) {
+  static void assertCloseCleanup(NonBlockingRouter router, long operationCount) {
     router.close();
     Assert.assertEquals("No ChunkFiller Thread should be running after the router is closed", 0,
         TestUtils.numThreadsByThisName("ChunkFillerThread"));
     Assert.assertEquals("No RequestResponseHandler should be running after the router is closed", 0,
         TestUtils.numThreadsByThisName("RequestResponseHandlerThread"));
-    Assert.assertEquals("All operations should have completed", 0, router.getOperationsCount());
+    long operationCountAfterTest = router.getOperationsCount();
+    Assert.assertEquals("All operations should have completed", operationCount, operationCountAfterTest);
```

Comment:
if all operations should have completed, why isn't the op count 0 ?

Reply:
Looks it will be non-zero if the other test is in the middle (not finished yet).

Reply:
We run tests serially as I recall. So we can't be in the middle of some test while running another.